yt_geotiff/io.py

```diff
@@ -127,6 +127,147 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
     
         return rv
 
+class rasterio_group(IOHandlerGeoTiff):
+    _dataset_type = "RasterioGroup"
+
+    # copy _read_fluid_selection
+    # Difference geotiff read specific band number
+    # 
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+
+        rv = {}
+        chunks = list(chunks)
+
+        if isinstance(selector, GridSelector):            
+
+            if not (len(chunks) == len(chunks[0].objs) == 1):
+                raise RuntimeError
+            g = chunks[0].objs[0]
+
+            if g.id in self._cached_fields:
+                gf = self._cached_fields[g.id]
+                rv.update(gf)
+
+            if len(rv) == len(fields):
+                return rv
+            
+            for field in fields:
+                if field in rv:
+                    self._hits += 1
+                    continue
+                self._misses += 1
+                ftype, fname = field 
+                
+                filename=self.ds.directory+'/'+self.ds._field_filename[fname]['filename']                              
+
+                src = rasterio.open(filename, "r")
+                rasterio_window = g._get_rasterio_window(selector, src.crs, src.transform)
+
+                # round up rasterio window width and height
+                rasterio_window = rasterio_window.round_shape(op='ceil', pixel_precision=None)
+
+                # Read in the band/field
+                data = src.read(1, window=rasterio_window).astype(
```

Comment:
If we are only reading band 1, then we don't support multi-band geotiffs in this dataset type. We should try to fix that soon.

Reply:
New dictionary introduced in data_structures.py (self.ds._file_band_number). Dictionary stores field names, full file paths and band number. If an image has more than one band (i.e. multiband) the field name is appended with "_BAND#".